piet/src/text.rs

```diff
@@ -28,7 +29,54 @@ pub trait TextLayoutBuilder {
     fn build(self) -> Result<Self::Out, Error>;
 }
 
+/// # Text Layout
+///
+/// A text position corresponds roughly to unicode code units. Because piet uses rust utf-8
+/// strings, this means that text position corresponods roughly with byte index.
+///
+/// However, text position is also related to valid cursor positions. Therefore:
+/// - The end of a line is a valid text position. `text.len()` is a valid text position.
+/// - If the text position is not at a code point or grapheme boundary, undesirable behavior may
+/// occur.
+///
 pub trait TextLayout {
     /// Measure the advance width of the text.
     fn width(&self) -> f64;
+
+    /// Given a Point, determine the corresponding text position
+    fn hit_test_point(&self, point: Point) -> HitTestPoint;
+
+    /// Given a text position, determine the corresponding pixel location
+    /// (currently consider the text layout just one line)
+    ///
+    /// Note: if text position is not at grapheme boundary, `cairo` and `web` will find the next text position/grapheme
+    /// boundary. This behavior is different than `directwrite`, which will panic.
```

Comment:
I posted this comment to Zulip also, but pasting here for posterity:

Really? I find that pretty hard to believe, and also super unfriendly behavior. Panic is OK if it's not at a *code point* boundary (btw, there's a [convention](https://rust-lang.github.io/api-guidelines/documentation.html#function-docs-include-error-panic-and-safety-considerations-c-failure) to use `# Panics` for this), but there are lots of legit ways a cursor can end up on a non-grapheme boundary. One I mentioned is pasting a lone [RIS](https://en.wikipedia.org/wiki/Regional_Indicator_Symbol) at the start of a RIS sequence.

Reply:
Fixed doc comment, plus improved handling of non-grapheme-boundary text positions.